### DISCUSSION TITLE
Packer Build With Managed Image

### DISCUSSION THREAD

**Discussion by kabal2010** [community/community#160667] [Actions]
Can we have a packer build that uses managed image as this will not require the creation of the associated storage account in Azure and the image been stored in VHD format? Also, if possible to have the packer build to have the capability of publishing the image to a shared image gallery.

I've been able to achieve both options, but had to customize the source code which means I'll only have a static manually adjusted packer configurations but it'll be helpful if the team can consider making then available.

**Comment by al-cheb** [comment]
Currently, we don't have any plans to migrate to managed disk. We will create a work item to this proposal.

**Reply by miketimofeev** [reply]
There was an issue and PR already, but we decided not to maintain managed images templates\scripts
https://github.com/actions/virtual-environments/pull/3955

**Reply by kabal2010** [reply]
Thanks

**Reply by sjesse** [reply]
FYI.. it appears Azure Packer Builder is looking to deprecate VHD and suggests moving to managed disks:
azure-arm: Warning: You are using Azure Packer Builder to create VHDs which is being deprecated, consider using Managed Images. Learn more https://www.packer.io/docs/builders/azure/arm#azure-arm-builder-specific-options

I agree with the OP, I have also been able to modify the scripts to use managed images and the shared image gallery which is our preferred method but I too do not like having to manually adjust this each time.

It would be nice to support managed Images at a minimum then the lift to move them to a shared image gallery could be smaller and a separate step in the pipeline.

**Reply by wsmelton** [reply]
Disappointing a tool owned and published by Microsoft refused to keep up with current technology. 😞 (or at least offer option for customers to if they wish)

**Comment by kabal2010** [comment]
This might not be the best place to raise this, but are you planning to add Terragrunt to the list of tools installed as it seems not part of the Linux builds?

**Comment by rybal06** [comment]
https://learn.microsoft.com/en-us/azure/virtual-machines/unmanaged-disks-deprecation

Microsoft has announced the below:
* As of November 18, 2022, new customer subscriptions won't be eligible to create unmanaged disks.
* As of September 30, 2023, existing customers won't be able to create new unmanaged disks.
* On September 30, 2025, customers will no longer be able to start IaaS VMs by using unmanaged disks. Any VMs that are still running or allocated will be stopped and deallocated.

I have tried converting the VHD file which packer outputs into an image gallery managed disk. This works for the Ubuntu images; but prevents the windows-2022 images from powering on.

Is there any roadmap for modifying the packer templates to output managed disks in an image gallery instead of a VHD in the storage account ahead of these deprecations?

**Reply by stozinho** [reply]
How did you do this for Ubuntu? Much obliged in advance

**Comment by egg-r** [comment]
@miketimofeev @al-cheb Wanted to get this back on your radar if you haven't seen this yet.  For the mentioned deprecation of unmanaged disks.  I just had to switch what I use for my "image factory" azure resources to a new subscription.  Apparently just made the cut, but others will not be so lucky soon.  I was looking forward to the mentioned updates to allow for managed disks as I wanted to also leverage an Azure Compute Gallery.  That's because I'm using these images in several self-hosted vmss pools in various regions and would auto-replicate it for me it looks like (haven't used this before).

**Comment by nnellanspdl** [comment]
To reiterate what was already said above, I am using the latest version of the build script (as of today) and I'm receiving this notice in the logs:

`==> azure-arm.build_vhd: Warning: You are using Azure Packer Builder to create VHDs which is being deprecated, consider using Managed Images.`
